components/fetch-build-scan-data-cmdline-tool/src/main/java/com/gradle/enterprise/api/client/GradleEnterpriseApiClient.java

```diff
@@ -99,6 +100,27 @@ private boolean allowUntrustedServer() {
         return Boolean.parseBoolean(System.getProperty("ssl.allowUntrustedServer"));
     }
 
+    private void configureTimeouts(OkHttpClient.Builder httpClientBuilder) {
+        Duration connectTimeout = parseTimeout("connect.timeout");
+        if (connectTimeout != null) {
+            httpClientBuilder.connectTimeout(connectTimeout);
+        }
+        Duration readTimeout = parseTimeout("read.timeout");
+        if (readTimeout != null) {
+            httpClientBuilder.readTimeout(readTimeout);
+        }
+    }
+
+    private Duration parseTimeout(String key) {
+        String value = System.getProperty(key);
+        if (value == null) {
+            // value is not set
+            return null;
+        } else {
+            return Duration.parse(key);
+        }
```

Comment:
I would use the ternary operator here to reduce this if/else to a single line.

Reply:
✅ Done